Z/ZPares/build_tarballs.jl

```diff
@@ -0,0 +1,103 @@
+    cp Makefile.inc/make.inc.gfortran.seq ./make.inc
+    make BLAS="-L$libdir -lopenblas" USE_MPI="0" FFLAG="-O3 $Rankmismatch -dynamiclib -L$libdir -lopenblas" LAPACK="-L./"
+    gfortran -O3 -dynamiclib -fPIC zpares_wrapper.f90 -I./include -L./lib -lzpares -L$libdir -lopenblas -o libzpares.dylib
+    cp libzpares.dylib lib/
+    cp lib/libzpares.dylib $prefix/lib/
+    cp include/zpares.mod $prefix/include/
+    cp zpares_wrapper.mod $prefix/include/
+    cp lib/libzpares.a $prefix/lib/
+elif [[ $target == *"w64-mingw32"* ]]; then 
+    cp Makefile.inc/make.inc.gfortran.seq ./make.inc
+    make BLAS="-L$libdir -lopenblas" LAPACK="-L$libdir -lopenblas" FFLAG="-O3 $Rankmismatch -shared -fPIC"
+    gfortran -O3 -shared -fPIC zpares_wrapper.f90 -I./include -L./lib -lzpares  -L$libdir -lopenblas -o zpares_wrapper.a
+    ld -shared -o zpares.so --whole-archive zpares_wrapper.a 
+    cp lib/* $prefix/bin/
+    cp zpares.so lib/
+    cp lib/zpares.so $prefix/lib/libzpares.dll
+    cp include/zpares.mod $prefix/include/
+    cp zpares_wrapper.mod $prefix/include/
+    cp lib/libzpares.a $prefix/lib/
+else
+    cp Makefile.inc/make.inc.gfortran.seq ./make.inc
+    make BLAS="-L$libdir -lopenblas" LAPACK="-L$libdir -lopenblas" FFLAG="-O3 $Rankmismatch -shared -fPIC"
+    gfortran -O3 -shared -fPIC zpares_wrapper.f90 -I./include -L./lib -lzpares  -o zpares_wrapper.a
+    ld -shared -o zpares.so --whole-archive zpares_wrapper.a 
+    cp zpares_wrapper.a $prefix/lib/
+    cp zpares.so lib/
+    cp lib/zpares.so $prefix/lib/libzpares.so
+    cp include/zpares.mod $prefix/include/
+    cp zpares_wrapper.mod $prefix/include/
+    cp lib/libzpares.a $prefix/lib/
+fi
+"""
+
+
+#    cp -r examples $prefix/
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "windows"; ),
+    Platform("i686", "windows"; ),
+    Platform("x86_64", "macos"; ),
+    Platform("aarch64", "macos"; ),
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("x86_64", "freebsd"; )
+]
+
+platforms = expand_gfortran_versions(platforms)
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libzpares", :libzpares)]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="OpenBLAS_jll", uuid="4536629a-c528-5b80-bd46-f80d51c5b363"))
+    Dependency(PackageSpec(name="OpenBLAS32_jll", uuid="656ef2d0-ae68-5445-9ca0-591084a874a2"))
```

Comment:
Why two different OpenBLASes?

Reply:
OpenBLAS_jll has libopenblas64_.so. OpenBLAS32_jll has libopenblas.so. I wanted to use the Lapack and BLAS to compile this package. But libopenblas64_.so can not be used. So I added OpenBLAS32_jll. I do not know the reason why we need OpenBLAS32_jll instead of OpenBLAS_jll. But it works. If we use OpenBLAS_jll only, the gfortran compiler can not find Lapack functions like zheev_. If you know the minimum requirement for using Lapack and BLAS in BinaryBuilder.jl, could you change it?

Reply:
> OpenBLAS_jll has libopenblas64_.so.

Yes, because it uses the 64-bit interface (ILP64) on 64-bit systems.

> OpenBLAS32_jll has libopenblas.so.

That has the 32-bit interface.

> I wanted to use the Lapack and BLAS to compile this package. But libopenblas64_.so can not be used.

Maybe because the package expects the 32-bit interface?

> So I added OpenBLAS32_jll.

So why adding OpenBLAS_jll too?

> I do not know the reason why we need OpenBLAS32_jll instead of OpenBLAS_jll.

Again, my guess is that the package expects the 32-bit interface.

>  If we use OpenBLAS_jll only, the gfortran compiler can not find Lapack functions like zheev_.

In the 64-bit interface used by Julia the symbols have `64_` suffix, so you'd need `zheev64_` instead of `zheev_`.  Note that you can rename symbols with macros, we have plenty of examples in this repository.

> If you know the minimum requirement for using Lapack and BLAS in BinaryBuilder.jl, could you change it?

Not sure what you mean.

To be clear, using 32-bit OpenBLAS is ok if that's what the package expects.  My question is about why using both.  I'm not sure that was answered.

Reply:
Thank you for replying. I understand what you meant. The answer is that we need OpenBLAS32_jll only. So I should remove OpenBLAS_jll.